lib/bridge/IrcHandler.js

```diff
@@ -201,6 +207,55 @@ IrcHandler.prototype.onMessage = Promise.coroutine(function*(req, server, fromUs
         return;
     }
 
+    // For topics, set matrix topic as AS bot user
+    if (action.type === "topic") {
+        // Only bridge topics for rooms created by the bridge, via !join or an alias
+        let origins = ["join", "alias"];
+        let mappings = yield this.ircBridge.getStore().getMappingsForChannelByOrigin(
+            server, channel, origins
+        );
+        let promises = mappings.map((entry) => {
+
+            if (this._roomIdToTopic[entry.matrix.getId()] === action.text) {
+                req.log.info(
+                    "Topic of %s already set to '%s'",
+                    entry.matrix.getId(),
+                    action.text
+                );
+                return;
+            }
+
+            req.log.info(
+                "Message is a topic, using AS bot to set topic in room %s: %s",
+                entry.matrix.getId(),
+                action.text
+            );
+
+            return this.ircBridge.getAppServiceBridge().getIntent().setRoomTopic(
```

Comment:
Yup. Of course, there is no guarantee that after we proclaim "the topic on room X is Y" that we will end up succeeding to set the topic... so a 3-state "it is X", "it is going to be Y", "it is Y" may be a better solution.